packages/fxa-shared/metrics/glean/web/index.ts

```diff
@@ -19,7 +19,14 @@ export type PropertyName = PropertyNameT[number];
 export type EventProperties = {
   [k in PropertyName]?: string;
 };
-export type EventMapKeys = keyof typeof eventsMap;
+export type EventsMap = typeof eventsMap;
+export type EventMapKeys = keyof EventsMap;
+
+// Type for optional values
+export type GleanPingMetrics = {
+  event?: EventProperties;
+  sync?: { cwts: Record<string, boolean> };
```

Comment:
Well we know the possible set of sync engine ids in the code base because they are hard coded in packages/fxa-settings/src/components/ChooseWhatToSync/sync-engines.ts.  We don't know from the `EngineConfig` type since the type of its `id` is just `string`.  (The actual engines available to the user as options on the UI is not static but I'm sure you didn't mean a runtime check.)  So, I'll try to export a sum type for the sync engine ids from sync-engines.ts and use it here.

Reply:
Actually how do you feel about fxa-shared importing from fxa-settings?

Reply:
Well I updated the types in sync-engines.ts but gonna hold off on importing the engine ids in fxa-shared.